homeassistant/components/tplink/switch.py

```diff
@@ -164,12 +168,27 @@ def update(self):
                 except KeyError:
                     # Device returned no daily history
                     pass
-
-            self._available = True
-
         except (SmartDeviceException, OSError) as ex:
+            _LOGGER.warning(
+                "Retrying in %s for %s|%s due to: %s",
+                SLEEP_TIME,
+                self.smartplug.host,
+                self._alias,
+                ex,
+            )
+            self._available = False
+        self._available = True
+
+    async def async_update(self):
+        """Update the TP-Link switch's state."""
+        for update_attempt in range(MAX_ATTEMPTS):
+            await self.hass.async_create_task(self.async_attempt_update())
```

Comment:
Schedule sync work on the executor thread pool with `self.hass.async_add_executor_job`.

Reply:
Switched to async_add_executor_job